logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt

```diff
@@ -80,6 +80,40 @@ class MLSConversationDataSource(
             establishMLSGroup(groupID, members)
         }
 
+    override suspend fun addMemberToMLSGroup(groupID: String, members: List<UserId>): Either<CoreFailure, Unit> =
+        //TODO: check for federated and non-federated members
+        keyPackageRepository.claimKeyPackages(members).flatMap { keyPackages ->
+            mlsClientProvider.getMLSClient().flatMap { client ->
+                val clientKeyPackageList = keyPackages
+                    .map {
+                        Pair(
+                            CryptoQualifiedClientId(it.clientID, CryptoQualifiedID(it.userId, it.domain)),
+                            it.keyPackage.decodeBase64Bytes()
+                        )
+                    }
+                client.addMember(groupID, clientKeyPackageList)?.let { (handshake, welcome) ->
+                    wrapApiRequest {
+                        mlsMessageApi.sendMessage(MLSMessageApi.Message(handshake))
+                    }.flatMap {
+                        wrapApiRequest {
+                            mlsMessageApi.sendWelcomeMessage(MLSMessageApi.WelcomeMessage(welcome))
+                        }
+                    }.flatMap {
+                        wrapStorageRequest {
+                            val list = members.map {
+                                com.wire.kalium.persistence.dao.Member(idMapper.toDaoModel(it))
+                            }
+                            conversationDAO.insertMembers(list, groupID)
+                        }
+                    }.flatMap {
+                        Either.Right(Unit)
+                    }
+                } ?: run {
+                    Either.Right(Unit)
```

Comment:
What does it mean in the value is null at the end, and why is null mapped to success? 
Maybe I'm missing some context here, some comments can clear stuff for every one

Reply:
We discussed that in your changes for admin users.